astropy/coordinates/builtin_frames/intermediate_rotation_transforms.py

```diff
@@ -162,16 +170,24 @@ def cirs_to_gcrs(cirs_coo, gcrs_frame):
     # compute the pmatrix, and then multiply by its transpose
     pmat = gcrs_to_cirs_mat(cirs_coo.obstime)
     newrepr = cirs_coo.cartesian.transform(matrix_transpose(pmat))
-    gcrs = GCRS(newrepr, obstime=cirs_coo.obstime)
-
-    # now do any needed offsets (no-op if same obstime and 0 pos/vel)
+    # can't use location.get_gcrs_posvel as that would trigger infinite recursion
+    obscirsloc, obscirsvel = cirs_coo.location.get_posvel(cirs_coo.obstime, CIRS)
+    obsgeoloc = obscirsloc.transform(matrix_transpose(pmat))
+    obsgeovel = obscirsvel.transform(matrix_transpose(pmat))
+    gcrs = GCRS(newrepr,
+                obstime=cirs_coo.obstime,
+                obsgeoloc=obsgeoloc,
+                obsgeovel=obsgeovel)
+
+    # now do any needed offsets (no-op if same obstime and pos/vel)
     return gcrs.transform_to(gcrs_frame)
 
 
 @frame_transform_graph.transform(FunctionTransformWithFiniteDifference, CIRS, ITRS)
 def cirs_to_itrs(cirs_coo, itrs_frame):
-    # first get us to CIRS at the target obstime
-    cirs_coo2 = cirs_coo.transform_to(CIRS(obstime=itrs_frame.obstime))
+    # first get us to geocentric CIRS at the target obstime
+    cirs_coo2 = cirs_coo.transform_to(CIRS(obstime=itrs_frame.obstime,
+                                           location=EarthLocation(0*u.km, 0*u.km, 0*u.km)))
```

Comment:
Is this ever likely to be a problem?

I'm not keen on the idea of static members of `EarthLocation` since that is going to get messy with the site registry - when should something be a static member and when in the site list for example. I have the same thoughts for the `ZERO_SPEED` constant.

Reply:
Maybe just define them as constants for `intermediate_rotation_transforms`? Or in `utils.py`, together with `DEFAULT_OBSTIME`? 

Although in this particular case, since the centre of the Earth is already the default, perhaps just omit it?

Reply:
done